numba/tests/test_parfors.py

```diff
@@ -2495,6 +2495,26 @@ def test_impl(x):
         x = [np.array([1,2,3], dtype=int),np.array([1,2], dtype=int)]
         self.prange_tester(test_impl, x)
 
+    @skip_parfors_unsupported
+    def test_ssa_false_reduction(self):
+        # issue5698
+        # SSA for h creates assignments to h that make it look like a
+        # reduction variable except that it lacks an associated
+        # reduction operator.  Test here that h is excluded as a
+        # reduction variable.
+        def test_impl(image, a):
+            empty = np.zeros(image.shape)
+            for i in range(image.shape[0]):
+                r = image[i][0] / 255.0
+                if a == 0:
+                    h = 0
+                empty[i] = [h, h, h]
+            return empty
```

Comment:
This doesn't seem to trip the bug, it works for me on master?! Minimal thing I have for hitting the bug is:
```
def test_impl(image, a, b):
    empty = np.zeros(image.shape)
    for i in prange(image.shape[0]):
        if b == 0:
            h = 0
        if a == 0:
            h = 0
        empty[i] = [h, h, h]
    return empty
```
both branches are needed.

Reply:
I think this might be so `h` gets versioned?

Reply:
I thought I tested this on master.

Reply:
If I put that test in a separate file and test on master then I still see the bug.  Wouldn't be the first time something behaved differently in the testing file versus one outside the testing infrastructure.  I'll change back to your code anyway.

Reply:
hmmm, strange, I get this:
```
$ git rev-parse --short upstream/master
64fbf2b82

$ git rev-parse --short HEAD
64fbf2b82

$ cat test_5705.py
from numba import njit, prange
import numpy as np

@njit(parallel=True)
def test_with_r(image, a, b):
    empty = np.zeros(image.shape)
    for i in prange(image.shape[0]):
        r = image[i][0] / 255.0
        if a == 0:
            h = 0
        empty[i] = [h, h, h]
    return empty

image = np.zeros((3, 3), dtype=np.int32)
test_with_r(image, 0, 0)

$ python test_5705.py

$
```
anyway, what's in this patch seems to pass/fail as expected so thanks for fixing.